src/main/webapp/app/exam/exam-scores/exam-scores.component.ts

```diff
@@ -185,88 +190,50 @@ export class ExamScoresComponent implements OnInit, OnDestroy {
-                    },
-                ],
-                xAxes: [
-                    {
-                        scaleLabel: {
-                            display: true,
-                            labelString: this.translateService.instant('artemisApp.examScores.xAxes'),
-                        },
-                    },
-                ],
-            },
-            hover: {
-                animationDuration: 0,
-            },
-            animation: {
-                duration: 1,
-                onComplete() {
-                    const chartInstance = this.chart,
-                        ctx = chartInstance.ctx;
-
-                    ctx.font = helpers.fontString(defaults.global.defaultFontSize, defaults.global.defaultFontStyle, defaults.global.defaultFontFamily);
-                    ctx.textAlign = 'center';
-                    ctx.textBaseline = 'bottom';
-
-                    this.data.datasets.forEach(function (dataset: DataSet, j: number) {
-                        const meta = chartInstance.controller.getDatasetMeta(j);
-                        meta.data.forEach(function (bar: any, index: number) {
-                            const data = dataset.data[index];
-                            ctx.fillText(data, bar._model.x, bar._model.y - 20);
-                            ctx.fillText(`(${component.roundAndPerformLocalConversion((data * 100) / component.noOfExamsFiltered)}%)`, bar._model.x, bar._model.y - 5);
-                        });
-                    });
-                },
-            },
-        };
 
-        this.barChartData = [
-            {
-                label: '# of students',
-                data: this.histogramData,
-                backgroundColor: 'rgba(0,0,0,0.5)',
-            },
-        ];
+        this.ngxData = [...this.ngxData];
+    }
+
+    /**
+     * Calculate statistics on exercise group and exercise granularity. These statistics are filter dependent.
+     * @param exerciseGroupResults Data structure holding the aggregated points and number of participants
+     */
+    private calculateExerciseGroupStatistics(exerciseGroupResults: AggregatedExerciseGroupResult[]) {
+        for (const groupResult of exerciseGroupResults) {
+            // For average points for exercise groups
+            if (groupResult.noOfParticipantsWithFilter) {
+                groupResult.averagePoints = groupResult.totalPoints / groupResult.noOfParticipantsWithFilter;
+                groupResult.averagePercentage = (groupResult.averagePoints / groupResult.maxPoints) * 100;
+                if (this.gradingScaleExists) {
+                    const gradeStep = this.gradingSystemService.findMatchingGradeStep(this.gradingScale!.gradeSteps, groupResult.averagePercentage);
+                    groupResult.averageGrade = gradeStep!.gradeName;
+                }
+            }
+            // Calculate average points for exercises
+            groupResult.exerciseResults.forEach((exResult) => {
```

Comment:
I would prefer having type here as well, i.e. `(exResult: [Type]) => ...`

Reply:
As far as I know, IntelliJ identifies the type itself and shows a little badge next to the attribute name. Actually this method is legacy code, I don't know really know why line changes are shown here. However, I explicitly stated the type in edf3ef8